### ISSUE TITLE
Empty geometries in polygon table

### ISSUE THREAD

**Issue by lonvia** [openstreetmap/osm2pgsql#292]
[This relation](https://www.openstreetmap.org/relation/3726445) leaves an entry with an empty geometry in the `planet_osm_polygon` table.

**Comment by mboeringa** [comment]
Why is way 169272321 added twice to the relation as outer? That seems a tagging error...

**Comment by pnorman** [comment]
this is probably the data that was causing it, since it's since been fixed in OSM

```xml
<?xml version='1.0' encoding='UTF-8'?>
<osm version='0.6' upload='true' generator='JOSM'>
  <node id='1804384035' timestamp='2012-06-30T12:13:58Z' uid='302852' user='be-ju' visible='true' version='2' changeset='12065158' lat='50.5539565' lon='8.6482181' />
  <node id='1804384038' timestamp='2012-06-30T12:13:59Z' uid='302852' user='be-ju' visible='true' version='2' changeset='12065158' lat='50.5539629' lon='8.6481648' />
  <node id='1804384039' timestamp='2012-06-30T12:13:59Z' uid='302852' user='be-ju' visible='true' version='2' changeset='12065158' lat='50.553966' lon='8.6481391' />
  <node id='1804384040' timestamp='2012-06-30T12:13:59Z' uid='302852' user='be-ju' visible='true' version='2' changeset='12065158' lat='50.5539686' lon='8.6481399' />
  <node id='1804384044' timestamp='2012-06-30T12:13:59Z' uid='302852' user='be-ju' visible='true' version='2' changeset='12065158' lat='50.5539795' lon='8.6482828' />
  <node id='1804384047' timestamp='2012-06-30T12:13:59Z' uid='302852' user='be-ju' visible='true' version='2' changeset='12065158' lat='50.5539862' lon='8.6482269' />
  <node id='1804384055' timestamp='2012-06-30T12:13:59Z' uid='302852' user='be-ju' visible='true' version='2' changeset='12065158' lat='50.5540076' lon='8.6481515' />
  <node id='1804384056' timestamp='2012-06-30T12:13:59Z' uid='302852' user='be-ju' visible='true' version='2' changeset='12065158' lat='50.5540093' lon='8.6481372' />
  <node id='1804384085' timestamp='2012-06-30T12:14:00Z' uid='302852' user='be-ju' visible='true' version='2' changeset='12065158' lat='50.5540642' lon='8.648308' />
  <node id='1804384087' timestamp='2014-05-08T13:55:12Z' uid='1837098' user='IkeH' visible='true' version='3' changeset='22209765' lat='50.5540702' lon='8.6481694' />
  <node id='1804384089' timestamp='2012-06-30T12:14:00Z' uid='302852' user='be-ju' visible='true' version='2' changeset='12065158' lat='50.5540725' lon='8.648156' />
  <node id='1804384092' timestamp='2012-06-30T12:14:00Z' uid='302852' user='be-ju' visible='true' version='2' changeset='12065158' lat='50.5540804' lon='8.6481732' />
  <way id='169272321' timestamp='2015-04-15T06:54:39Z' uid='2230854' user='RadFr' visible='true' version='7' changeset='30228506'>
    <nd ref='1804384056' />
    <nd ref='1804384089' />
    <nd ref='1804384087' />
    <nd ref='1804384055' />
    <nd ref='1804384056' />
    <tag k='building' v='yes' />
  </way>
  <way id='169272605' timestamp='2015-04-15T06:54:39Z' uid='2230854' user='RadFr' visible='true' version='7' changeset='30228506'>
    <nd ref='1804384038' />
    <nd ref='1804384035' />
    <nd ref='1804384047' />
    <nd ref='1804384044' />
    <nd ref='1804384085' />
    <nd ref='1804384092' />
    <nd ref='1804384087' />
    <nd ref='1804384055' />
    <nd ref='1804384040' />
    <nd ref='1804384039' />
    <nd ref='1804384038' />
    <tag k='addr:city' v='Kleinlinden' />
    <tag k='addr:country' v='DE' />
    <tag k='addr:housenumber' v='12' />
    <tag k='addr:postcode' v='35398' />
    <tag k='addr:state' v='Hessen' />
    <tag k='addr:street' v='Gregor-Mendel-Straße' />
    <tag k='building' v='yes' />
    <tag k='source' v='HiRes aerial imagery (Aerowest)' />
  </way>
  <relation id="3726445" changeset="29447760" timestamp="2015-03-13T13:16:42Z" version="2" visible="true" user="KartoGrapHiti" uid="57645">
    <member type="way" ref="169272321" role="outer"/>
    <member type="way" ref="169272605" role="outer"/>
    <member type="way" ref="169272321" role="outer"/>
    <tag k="addr:city" v="Kleinlinden"/>
    <tag k="addr:country" v="DE"/>
    <tag k="addr:housenumber" v="12"/>
    <tag k="addr:postcode" v="35398"/>
    <tag k="addr:state" v="Hessen"/>
    <tag k="addr:street" v="Gregor-Mendel-Straße"/>
    <tag k="building" v="yes"/>
    <tag k="source" v="HiRes aerial imagery (Aerowest)"/>
    <tag k="type" v="multipolygon"/>
  </relation>
</osm>
```

**Comment by lonvia** [comment]
Fixed by #655.
